src/lib/routes/client-api/feature.ts

```diff
@@ -50,20 +58,34 @@ export default class FeatureController extends Controller {
         }
     }
 
-    async getAll(req: Request, res: Response): Promise<void> {
-        const query = await this.prepQuery(req.query);
-        let features;
-        if (this.cache) {
-            features = await this.cachedFeatures(query);
-        } else {
-            features = await this.featureToggleServiceV2.getClientFeatures(
-                query,
-            );
+    private async resolveQuery(
+        req: IAuthRequest,
+    ): Promise<IFeatureToggleQuery> {
+        const { user, query } = req;
+
+        const override: QueryOverride = {};
+        if (user instanceof ApiUser) {
+            if (user.project !== ALL) {
+                override.project = [user.project];
+            }
+            if (user.environment !== ALL) {
+                override.environment = user.environment;
+            }
```

Comment:
Ah, there's the solution for * being all, don't include it in the query.

Reply:
jup. If the token is only for a specific env/project that will be selected always. If the key allow "ALL" you can fetch all or scope your query with regular query arguments.